### PR TITLE
feat: Kerberos authentication for MQTT clients

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-11630

Related https://github.com/kafka4beam/sasl_auth/pull/29

Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_conf/include/emqx_conf.hrl [hunk 1]
```diff
@@ -37,4 +37,61 @@
 
 -define(READONLY_KEYS, [cluster, rpc, node]).
 
+-define(CE_AUTHZ_SOURCE_SCHEMA_MODS, [
```
Comment (zmstone) [inline apps/emqx_conf/include/emqx_conf.hrl:40]: this was moved  to`emqx_conf_schema_inject`, why added back ?

Comment (zhongwencool) [reply apps/emqx_conf/include/emqx_conf.hrl]: ok

File: apps/emqx_auth_gssapi/include/emqx_auth_gssapi.hrl [hunk 2]
```diff
@@ -0,0 +1,28 @@
+%%--------------------------------------------------------------------
```
Comment (zmstone) [inline apps/emqx_auth_gssapi/include/emqx_auth_gssapi.hrl:1]: bsl

File: apps/emqx_conf/src/emqx_conf_schema_inject.erl [hunk 3]
```diff
@@ -51,7 +51,8 @@ authn_mods(ce) ->
         emqx_authn_http_schema,
         emqx_authn_jwt_schema,
         emqx_authn_scram_mnesia_schema,
-        emqx_authn_ldap_schema
+        emqx_authn_ldap_schema,
+        emqx_authn_gssapi_schema
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_conf_schema_inject.erl:55]: should be for ee only

File: apps/emqx_auth_gssapi/src/emqx_authn_gssapi.erl [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+]).
+
+create(
+    AuthenticatorID,
+    #{
+        principal := Principal,
+        keytab_file := KeyTabFile
+    }
+) ->
+    case sasl_auth:kinit(KeyTabFile, Principal) of
```
Comment (zmstone) [inline apps/emqx_auth_gssapi/src/emqx_authn_gssapi.erl:39]:
```suggestion
    Path = emqx_schema:naive_env_interpolation(KeyTabFile),
    case sasl_auth:kinit(Path, Principal) of
```

File: apps/emqx_auth_gssapi/src/emqx_authn_gssapi.erl [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            sasl_auth:server_done(SaslConn),
+            {error, not_authorized}
+    end.
+
+auth_continue(SaslConn, ClientToken) ->
+    case sasl_auth:server_step(SaslConn, ClientToken) of
+        {ok, {sasl_continue, ServerToken}} ->
+            {continue, ServerToken, #{sasl_conn => SaslConn}};
+        {ok, {sasl_ok, ServerToken}} ->
+            sasl_auth:server_done(SaslConn),
```
Comment (zmstone) [inline apps/emqx_auth_gssapi/src/emqx_authn_gssapi.erl:102]: call done when error as well ?

**Comment by zmstone** [comment]
will followup from here: https://github.com/emqx/emqx/pull/13601

**Pull request closed without merging**